### ISSUE TITLE
Gitpod resources are wasted by running multiple prebuild processes for same commit

### ISSUE THREAD

**Issue by shaal** [gitpod-io/gitpod#14237]
### Bug description

Pushing a commit result in 3 (three!) separate prebuild processes for no apparent reason.
![image](https://user-images.githubusercontent.com/22901/198302466-0ab334aa-b445-4664-81d5-8a25f136adbc.png)

https://gitpod.io/t/phase2/tufts-core/f3835ce4-aa28-4ab6-975a-7dc95d129e83
https://gitpod.io/t/phase2/tufts-core/f5862986-03e7-4268-87ea-fcaac8236931
https://gitpod.io/t/phase2/tufts-core/552b3fc5-c58f-4d23-ac52-af2ea71fd9b6

### Steps to reproduce

1. Add a repo to team/project, with gitpod.yml setup for Github prebuild.
1. Push a commit
1. Check the prebuild pages and see that multiple prebuilds are running (usually 2)
`https://gitpod.io/t/team/project/prebuilds`

**Comment by AlexTugarev** [comment]
Thanks for reporting @shaal!

This is not what we expect to happen. I did a quick look at the webhook event registered, and see 3 distinct events:

```
|creationTime            |type        |authorizedUserId                    |rawEvent             |
|------------------------|------------|------------------------------------|---------------------|
|2022-10-27T13:46:47.363Z|pull_request|A                                   |                     |
|2022-10-27T13:46:45.764Z|push        |A                                   |w/ installation id   |
|2022-10-27T13:46:45.592Z|push        |B                                   |w/o installation id  |

```

The `pull_request` event is slipped through despite https://github.com/gitpod-io/gitpod/pull/13295, which is reasonable if different server pods handled the requests. 

The new issue here is, the two `push` events look like they've been produced by two webhooks. While one is from the GH App, where is origin of the second event?

@shaal, do you have a change to inspect the Webhooks page for the `phase2/tufts-core` repository?

**Comment by AlexTugarev** [comment]
Following the current trend, a holistic way to solve this issue would be to move triggering of prebuilds into a control loop which would query the DB for recently recorded webhook events.

Also `pull_request` events should not trigger a prebuild at all. To clarify, there are two types of PRs which would be registered in those events. If the source repository is the same as the current target, a `push` event would also be registered. If the source is a different repository, we'd likely expect Gitpod would be configured there as well.

**Comment by shaal** [comment]
@AlexTugarev I noticed this is still happening
![image](https://user-images.githubusercontent.com/22901/216072545-4ab90b83-fc10-48ce-8a1d-97bf935fd664.png)

Where should I look?
The example in the screenshot is from a branch with no PR.
I pushed code to Github, and I see that 2 prebuilds are running in parallel.

**Comment by Konard** [comment]
![image](https://github.com/gitpod-io/gitpod/assets/1431904/75410de1-9ff4-483f-b968-fa4ba241eaf1)

I have the same issue again, please reopen the issue.

**Comment by Konard** [comment]
this repeats again

![IMG_1685](https://github.com/gitpod-io/gitpod/assets/1431904/14b1383d-5eb3-40df-bf70-6f943127a820)

**Comment by loujaybee** [comment]
[Related internal issue](https://linear.app/gitpod/issue/EXP-843/deduplicate-prebuilds-using-a-mutex-lock).
